include/boost/geometry/util/range.hpp

```diff
@@ -107,25 +107,13 @@ at(RandomAccessRange & rng,
 \ingroup utility
 */
 template <typename Range>
-inline typename boost::range_value<Range>::type const&
+inline typename boost::range_reference<Range>::type
 front(Range const& rng)
 {
     BOOST_ASSERT(!boost::empty(rng));
     return *boost::begin(rng);
 }
 
-/*!
-\brief Short utility to conveniently return the front element of a Range.
-\ingroup utility
-*/
-template <typename Range>
-inline typename boost::range_value<Range>::type &
-front(Range & rng)
-{
-    BOOST_ASSERT(!boost::empty(rng));
-    return *boost::begin(rng);
-}
-
```

Comment:
`front()` taking non-const reference shouldn't be removed. With your change it isn't possible to return a non-const reference if a non-const Container is passed because it'd be passed as const into the function. Consider:

    std::vector<int> v(10, 0);
    int & i = range::front(v);